### ISSUE TITLE
parser "recovers" after failing to parse type of fn arg

### ISSUE THREAD

**Issue by durka** [rust-lang/rust#39651]
If the type of a function argument doesn't parse, the parser doesn't stop but continues as if the argument never existed. This causes bogus errors later on about wrong number of arguments.

Ex:

```rust
fn foo(x: ????????) {} //~ERROR expected type, found `?`

foo(42); //~ERROR this function takes 0 parameters
```

**Comment by durka** [comment]
This doesn't seem to be an issue anymore.
